src/array/cpu/disjoint_union.cc

```diff
@@ -85,32 +85,38 @@ COOMatrix DisjointUnionCoo(const std::vector<COOMatrix>& coos) {
   auto res_dst_data = result_dst.Ptr<IdType>();
   auto res_dat_data = result_dat.Ptr<IdType>();
 
-  dgl::runtime::parallel_for(0, coos.size(), [&](IdType b, IdType e) {
-    for (IdType i = b; i < e; ++i) {
-      const aten::COOMatrix& coo = coos[i];
-      if (!coo.row_sorted) row_sorted = false;
-      if (!coo.col_sorted) col_sorted = false;
-
-      auto edges_src = coo.row.Ptr<IdType>();
-      auto edges_dst = coo.col.Ptr<IdType>();
-      auto edges_dat = coo.data.Ptr<IdType>();
-
-      for (IdType j = 0; j < coo.row->shape[0]; j++) {
-        res_src_data[prefix_elm[i] + j] = edges_src[j] + prefix_src[i];
-      }
-
-      for (IdType j = 0; j < coo.row->shape[0]; j++) {
-        res_dst_data[prefix_elm[i] + j] = edges_dst[j] + prefix_dst[i];
-      }
-
-      if (has_data) {
-        for (IdType j = 0; j < coo.row->shape[0]; j++) {
-          const auto d = (!COOHasData(coo)) ? j : edges_dat[j];
-          res_dat_data[prefix_elm[i] + j] = d + prefix_elm[i];
+  // 32 is a number obtained from experience. If a user set the grain size
+  // explicitly via env, use that value(`default_grain_size`) instead.
+  size_t grain_size = std::getenv("DGL_PARALLEL_FOR_GRAIN_SIZE")
+                          ? runtime::default_grain_size()
+                          : 32;
```

Comment:
Consider creating a ./include/env_variable.h, and include the header here, instead of calling std::getenv everywhere, which makes the output hard to predict.

```
static const char* kDGLParallelForGrainSize = std::getenv("DGL_PARALLEL_FOR_GRAIN_SIZE");
```

Reply:
make sense, changed for fix.